include/deal.II/multigrid/mg_constrained_dofs.h

```diff
@@ -146,13 +168,53 @@ MGConstrainedDoFs::initialize(const DoFHandler<dim,spacedim> &dof)
 {
   boundary_indices.clear();
   refinement_edge_indices.clear();
+  level_constraints.clear();
 
   const unsigned int nlevels = dof.get_triangulation().n_global_levels();
 
-  //At this point refinement_edge_indices is empty.
+  //At this point level_constraint and refinement_edge_indices are empty.
+  level_constraints.resize(nlevels);
   refinement_edge_indices.resize(nlevels);
   for (unsigned int l=0; l<nlevels; ++l)
-    refinement_edge_indices[l] = IndexSet(dof.n_dofs(l));
+    {
+      IndexSet relevant_dofs;
+      DoFTools::extract_locally_relevant_level_dofs(dof,l,relevant_dofs);
+      level_constraints[l].reinit(relevant_dofs);
```

Comment:
I added a line to continue if `cell->periodic_neighbor(f)->level_subdomain_id() == numbers::artificial_subdomain_id` returns true. The test passes and it doesn't affect my step 37 computation.